### PR TITLE
Added Fabric Error Counters and API

### PR DESCRIPTION
**Pull request by vaidyan1-int**
Re-instated the Fabric error counters and API for 1.7

### CODE REVIEW

File: scripts/sysman/fabric.yml [hunk 1]
```diff
@@ -185,6 +185,26 @@ members:
       name: "txCounter"
       desc: "[out] Monotonic counter for the number of bytes transmitted (sum of all lanes). This includes all protocol overhead, not only the GPU traffic."
 --- #--------------------------------------------------------------------------
+type: struct
+desc: "Fabric Port Error Counters"
+class: $sFabricPort
+name: $s_fabric_port_error_counters_t
+base: $s_base_properties_t
```
Comment (wdamon-intel) [inline scripts/sysman/fabric.yml:192]: Please add the corresponding `FABRIC_PORT_ERROR_COUNTERS` enumeration to the `$s_structure_type_t` enum.

**Review by wdamon-intel — CHANGES_REQUESTED** [review]
Looks good overall, but please address the structure type enumeration comment.

File: scripts/sysman/fabric.yml (hunk 1 above)
Comment (vaidyan1-int) [reply scripts/sysman/fabric.yml]: Updated strucutre type and added that to the enum in common.yml

File: scripts/sysman/fabric.yml [hunk 2]
```diff
@@ -185,6 +185,26 @@ members:
       name: "txCounter"
       desc: "[out] Monotonic counter for the number of bytes transmitted (sum of all lanes). This includes all protocol overhead, not only the GPU traffic."
 --- #--------------------------------------------------------------------------
+type: struct
+desc: "Fabric Port Error Counters"
+class: $sFabricPort
+name: $s_fabric_port_error_counters_t
+base: $s_structure_type_t
```
Comment (wdamon-intel) [inline scripts/sysman/fabric.yml:192]: `$s_structure_type_t` is an enum and cannot be the base type. As the members of this struct are properties of the fabric port error counters `$s_base_properties_t` would be the appropriate base type and would provide the extendibility of having an `stype` and `pNext`.

**Review by wdamon-intel — CHANGES_REQUESTED** [review]
Please address the type comment.

File: scripts/sysman/fabric.yml (hunk 2 above)
Comment (vaidyan1-int) [reply scripts/sysman/fabric.yml]: Fixed.

**Review by wdamon-intel — COMMENTED** [review]
LGTM

@jandres742  what do you think?

**Review by wdamon-intel — APPROVED** [review]
Marking as Approved for me.

**Review by jandres742 — APPROVED** [review]

**Pull request merged**